src/plugins/mer/merlocalhostavailableports.h

```diff
@@ -0,0 +1,62 @@
+/****************************************************************************
+**
+** Copyright (C) 2020 Open Mobile Platform LLC.
+** Contact: http://jolla.com/
+**
+** This file is part of Qt Creator.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** Other Usage
+**
+** Alternatively, this file may be used in accordance with the terms and
+** conditions contained in a signed written agreement between you and Digia.
+**
+****************************************************************************/
+
+#ifndef MERLOCALHOSTAVAILABLEPORTS_H
+#define MERLOCALHOSTAVAILABLEPORTS_H
+
+#include <projectexplorer/devicesupport/deviceusedportsgatherer.h>
+
+namespace Mer {
+namespace Internal {
+
+class MerLocalhostAvailablePorts : public QObject
+{
+    Q_OBJECT
+    Q_DISABLE_COPY(MerLocalhostAvailablePorts)
+
+public:
+    enum State { Invalid, Refresing, Valid, Error };
+    Q_ENUM(State)
+
+    ~MerLocalhostAvailablePorts() override;
+    static MerLocalhostAvailablePorts& instance();
```

Comment:
Rarely needed instance, I'd prefer using shared pointers to keep it around only when used.

Reply:
I don't get it. Do you propose omit "static" or return shared pointer?

Reply:
I mean something like

```
    static QSharedPointer<MerLocalhostAvailablePorts> instance();
...
private:
    static QWeakPointer<MerLocalhostAvailablePorts> s_instance;
```

Reply:
But more convenient would be to de-singletonize this class and instead use single ProjectExplorer::DeviceUsedPortsGatherer shared by all instances of this class.

Reply:
Ok, I understand what I should do. But I still not get where is a convenience. Using your approach I will be forced to instantiate MerLocalhostAvailablePorts in [merbuildenginedetailswidget.h](https://github.com/sailfishos/sailfish-qtcreator/blob/c8251ef1910756cbf88646c8b128529253881b2a/src/plugins/mer/merbuildenginedetailswidget.h) and [meremulatordetailswidget.h](https://github.com/sailfishos/sailfish-qtcreator/blob/c8251ef1910756cbf88646c8b128529253881b2a/src/plugins/mer/meremulatordetailswidget.h). Is this ok?

Reply:
Yes and both instances will use single shared instance of DeviceUsedPortsGatherer. Both efficient and convenient to use.

Reply:
PR updated. Now there is no static instance.